src/designspace.rs

```diff
@@ -0,0 +1,212 @@
+    pub minimum: Option<f32>,
+    /// The maximum value for a continuous axis, in user space coordinates.
+    pub maximum: Option<f32>,
+    /// The possible values for a discrete axis, in user space coordinates.
+    pub values: Option<Vec<f32>>,
+    /// Mapping between user space coordinates and design space coordinates.
+    pub map: Option<Vec<AxisMapping>>,
+}
+
+/// Maps one input value (user space coord) to one output value (design space coord).
+#[derive(Clone, Debug, Default, PartialEq, Deserialize)]
+#[serde(rename = "map")]
+pub struct AxisMapping {
+    /// user space coordinate
+    pub input: f32,
+    /// designspace coordinate
+    pub output: f32,
+}
+
+/// https://fonttools.readthedocs.io/en/latest/designspaceLib/xml.html#sources-element
+#[derive(Clone, Debug, Default, PartialEq, Deserialize)]
+#[serde(rename = "sources")]
+pub struct Sources {
+    /// One or more sources.
+    pub source: Vec<Source>,
+}
+
+/// A [source]].
+///
+/// [source]: https://fonttools.readthedocs.io/en/latest/designspaceLib/xml.html#id25
+#[derive(Clone, Debug, Default, PartialEq, Deserialize)]
+#[serde(rename = "source")]
+pub struct Source {
+    /// The family name of the source font.
+    pub familyname: Option<String>,
+    /// The style name of the source font.
+    pub stylename: Option<String>,
+    /// A unique name that can be used to identify this font if it needs to be referenced elsewhere.
+    pub name: String,
+    /// A path to the source file, relative to the root path of this document. The path can be at the same level as the document or lower.
+    pub filename: String,
+    /// The name of the layer in the source file. If no layer attribute is given assume the foreground layer should be used.
+    pub layer: Option<String>,
+    /// Location in designspace coordinates.
+    pub location: Location,
+}
+
+/// https://fonttools.readthedocs.io/en/latest/designspaceLib/xml.html#instances-element
+#[derive(Clone, Debug, Default, PartialEq, Deserialize)]
+#[serde(rename = "instances")]
+pub struct Instances {
+    /// One or more instances located somewhere in designspace.
+    pub instance: Vec<Instance>,
+}
+
+/// An [instance]].
+///
+/// [instance]: https://fonttools.readthedocs.io/en/latest/designspaceLib/xml.html#instance-element
+#[derive(Clone, Debug, Default, PartialEq, Deserialize)]
+pub struct Instance {
+    // per @anthrotype, contrary to spec, familyname and stylename are optional
+    /// The family name of the instance font. Corresponds with font.info.familyName
+    pub familyname: Option<String>,
+    /// The style name of the instance font. Corresponds with font.info.styleName
+    pub stylename: Option<String>,
+    /// A unique name that can be used to identify this font if it needs to be referenced elsewhere.
+    pub name: String,
+    /// A path to the instance file, relative to the root path of this document. The path can be at the same level as the document or lower.
+    pub filename: String,
```

Comment:
this should be optional too, only makes sense when interpolating static UFO instances, for defining VF named instances the output filename is not used and should not be required. Spec needs updating too

Reply:
Done, ty